Edit description from Integration Summary returns 400

```bash
8-02-07 10:26:45.648 ERROR [-,76e0e2a9dc219564,76e0e2a9dc219564,false] 1 --- [  XNIO-3 task-6] i.s.r.v.h.e.ReaderExceptionMapper        : com.fasterxml.jackson.databind.JsonMappingException: Unexpected token (END_OBJECT), expected FIELD_NAME: missing property 'actionType' that is to contain type id  (for class io.syndesis.model.action.Action)

  | at [Source: io.undertow.servlet.spec.ServletInputStreamImpl@78d79dd4; line: 31, column: 7] (through reference chain: io.syndesis.model.integration.Integration$Builder["steps"]->java.util.ArrayList[0]->io.syndesis.model.integration.Step$Builder["action"])
  |  
  | org.jboss.resteasy.spi.ReaderException: com.fasterxml.jackson.databind.JsonMappingException: Unexpected token (END_OBJECT), expected FIELD_NAME: missing property 'actionType' that is to contain type id  (for class io.syndesis.model.action.Action)
  | at [Source: io.undertow.servlet.spec.ServletInputStreamImpl@78d79dd4; line: 31, column: 7] (through reference chain: io.syndesis.model.integration.Integration$Builder["steps"]->java.util.ArrayList[0]->io.syndesis.model.integration.Step$Builder["action"])
  | at org.jboss.resteasy.core.MessageBodyParameterInjector.inject(MessageBodyParameterInjector.java:180) ~[resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
  | at org.jboss.resteasy.core.MethodInjectorImpl.injectArguments(MethodInjectorImpl.java:92) ~[resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
  | at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:115) ~[resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
  | at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:294) ~[resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
  | at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:248) ~[resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
  | at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:235) ~[resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
  | at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:402) [resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
  | at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:209) [resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
  | at org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.service(ServletContainerDispatcher.java:227) [resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
  | at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:56) [resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
  | at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:51) [resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
  | at javax.servlet.http.HttpServlet.service(HttpServlet.java:790) [javax.servlet-api-3.1.0.jar!/:3.1.0]
  | at io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:85) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:129) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at org.springframework.boot.web.filter.ApplicationContextHeaderFilter.doFilterInternal(ApplicationContextHeaderFilter.java:55) [spring-boot-1.5.8.RELEASE.jar!/:1.5.8.RELEASE]
  | at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
  | at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at org.springframework.web.filter.CorsFilter.doFilterInternal(CorsFilter.java:96) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
  | at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
  | at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at org.springframework.boot.actuate.trace.WebRequestTraceFilter.doFilterInternal(WebRequestTraceFilter.java:110) [spring-boot-actuator-1.5.8.RELEASE.jar!/:1.5.8.RELEASE]
  | at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
  | at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:317) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:127) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:91) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:114) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:137) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:111) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:111) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:170) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:63) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.security.web.authentication.preauth.AbstractPreAuthenticatedProcessingFilter.doFilter(AbstractPreAuthenticatedProcessingFilter.java:121) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:116) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:64) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
  | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:105) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:56) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
  | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:214) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:177) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
  | at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:347) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
  | at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:263) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
  | at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:99) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
  | at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
  | at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at org.springframework.web.filter.HttpPutFormContentFilter.doFilterInternal(HttpPutFormContentFilter.java:108) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
  | at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
  | at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at org.springframework.web.filter.HiddenHttpMethodFilter.doFilterInternal(HiddenHttpMethodFilter.java:81) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
  | at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
  | at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at org.springframework.cloud.sleuth.instrument.web.TraceFilter.doFilter(TraceFilter.java:186) [spring-cloud-sleuth-core-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
  | at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:197) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
  | at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
  | at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at org.springframework.boot.actuate.autoconfigure.MetricsFilter.doFilterInternal(MetricsFilter.java:106) [spring-boot-actuator-1.5.8.RELEASE.jar!/:1.5.8.RELEASE]
  | at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
  | at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.handlers.FilterHandler.handleRequest(FilterHandler.java:84) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:62) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.handlers.ServletChain$1.handleRequest(ServletChain.java:64) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:131) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:57) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43) [undertow-core-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:46) [undertow-core-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:64) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:60) [undertow-core-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:77) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43) [undertow-core-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43) [undertow-core-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43) [undertow-core-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:292) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.handlers.ServletInitialHandler.access$100(ServletInitialHandler.java:81) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:138) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:135) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:48) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:272) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.handlers.ServletInitialHandler.access$000(ServletInitialHandler.java:81) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.servlet.handlers.ServletInitialHandler$1.handleRequest(ServletInitialHandler.java:104) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.server.Connectors.executeRootHandler(Connectors.java:332) [undertow-core-1.4.21.Final.jar!/:1.4.21.Final]
  | at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:830) [undertow-core-1.4.21.Final.jar!/:1.4.21.Final]
  | at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [na:1.8.0_141]
  | at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [na:1.8.0_141]
  | at java.lang.Thread.run(Thread.java:748) [na:1.8.0_141]
  | Caused by: com.fasterxml.jackson.databind.JsonMappingException: Unexpected token (END_OBJECT), expected FIELD_NAME: missing property 'actionType' that is to contain type id  (for class io.syndesis.model.action.Action)
  | at [Source: io.undertow.servlet.spec.ServletInputStreamImpl@78d79dd4; line: 31, column: 7] (through reference chain: io.syndesis.model.integration.Integration$Builder["steps"]->java.util.ArrayList[0]->io.syndesis.model.integration.Step$Builder["action"])
  | at com.fasterxml.jackson.databind.JsonMappingException.from(JsonMappingException.java:270) ~[jackson-databind-2.8.10.jar!/:2.8.10]
  | at com.fasterxml.jackson.databind.DeserializationContext.wrongTokenException(DeserializationContext.java:1376) ~[jackson-databind-2.8.10.jar!/:2.8.10]
  | at com.fasterxml.jackson.databind.DeserializationContext.reportWrongTokenException(DeserializationContext.java:1197) ~[jackson-databind-2.8.10.jar!/:2.8.10]
  | at com.fasterxml.jackson.databind.jsontype.impl.AsPropertyTypeDeserializer._deserializeTypedUsingDefaultImpl(AsPropertyTypeDeserializer.java:165) ~[jackson-databind-2.8.10.jar!/:2.8.10]
  | at com.fasterxml.jackson.databind.jsontype.impl.AsPropertyTypeDeserializer.deserializeTypedFromObject(AsPropertyTypeDeserializer.java:105) ~[jackson-databind-2.8.10.jar!/:2.8.10]
  | at com.fasterxml.jackson.databind.deser.AbstractDeserializer.deserializeWithType(AbstractDeserializer.java:209) ~[jackson-databind-2.8.10.jar!/:2.8.10]
  | at com.fasterxml.jackson.datatype.jdk8.OptionalDeserializer.deserialize(OptionalDeserializer.java:101) ~[jackson-datatype-jdk8-2.8.10.jar!/:2.8.10]
  | at com.fasterxml.jackson.datatype.jdk8.OptionalDeserializer.deserialize(OptionalDeserializer.java:13) ~[jackson-datatype-jdk8-2.8.10.jar!/:2.8.10]
  | at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:504) ~[jackson-databind-2.8.10.jar!/:2.8.10]
  | at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeSetAndReturn(MethodProperty.java:116) ~[jackson-databind-2.8.10.jar!/:2.8.10]
  | at com.fasterxml.jackson.databind.deser.BuilderBasedDeserializer.vanillaDeserialize(BuilderBasedDeserializer.java:266) ~[jackson-databind-2.8.10.jar!/:2.8.10]
  | at com.fasterxml.jackson.databind.deser.BuilderBasedDeserializer.deserialize(BuilderBasedDeserializer.java:154) ~[jackson-databind-2.8.10.jar!/:2.8.10]
  | at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:287) ~[jackson-databind-2.8.10.jar!/:2.8.10]
  | at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:259) ~[jackson-databind-2.8.10.jar!/:2.8.10]
  | at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:26) ~[jackson-databind-2.8.10.jar!/:2.8.10]
  | at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:504) ~[jackson-databind-2.8.10.jar!/:2.8.10]
  | at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeSetAndReturn(MethodProperty.java:116) ~[jackson-databind-2.8.10.jar!/:2.8.10]
  | at com.fasterxml.jackson.databind.deser.BuilderBasedDeserializer.vanillaDeserialize(BuilderBasedDeserializer.java:266) ~[jackson-databind-2.8.10.jar!/:2.8.10]
  | at com.fasterxml.jackson.databind.deser.BuilderBasedDeserializer.deserialize(BuilderBasedDeserializer.java:154) ~[jackson-databind-2.8.10.jar!/:2.8.10]
  | at com.fasterxml.jackson.databind.ObjectReader._bind(ObjectReader.java:1583) ~[jackson-databind-2.8.10.jar!/:2.8.10]
  | at com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:964) ~[jackson-databind-2.8.10.jar!/:2.8.10]
  | at org.jboss.resteasy.plugins.providers.jackson.ResteasyJackson2Provider.readFrom(ResteasyJackson2Provider.java:134) ~[resteasy-jackson2-provider-3.1.4.Final.jar!/:3.1.4.Final]
  | at org.jboss.resteasy.core.interception.jaxrs.AbstractReaderInterceptorContext.readFrom(AbstractReaderInterceptorContext.java:66) ~[resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
  | at org.jboss.resteasy.core.interception.jaxrs.ServerReaderInterceptorContext.readFrom(ServerReaderInterceptorContext.java:61) ~[resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
  | at org.jboss.resteasy.core.interception.jaxrs.AbstractReaderInterceptorContext.proceed(AbstractReaderInterceptorContext.java:56) ~[resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
  | at org.jboss.resteasy.core.MessageBodyParameterInjector.inject(MessageBodyParameterInjector.java:147) ~[resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
  | ... 110 common frames omitted
```

--- a/app/ui/src/app/integration/detail-page/detail.component.html
+++ b/app/ui/src/app/integration/detail-page/detail.component.html
@@ -23,11 +23,25 @@
       </div>
     </div>
     <div class="col-md-12 details">
-      <h1>
+      <h1 class="inline-block">
         <syndesis-editable-text [value]="integration.name"
                                 [validationFn]="validateName"
                                 (onSave)="attributeUpdated('name', $event, integration)"></syndesis-editable-text>
       </h1>
+      <ng-container [ngSwitch]="integration.currentState">
+        <ng-container *ngSwitchCase="'Pending'">
+          <div class="spinner spinner-sm spinner-inline"></div> Publishing...
+        </ng-container>
+        <ng-container *ngSwitchCase="'Published'">
+          <span class="pficon pficon-ok"></span> Published version {{ integration.deploymentVersion }}
+        </ng-container>
+        <ng-container *ngSwitchCase="'Unpublished'">
+          Not Published
+        </ng-container>
+        <ng-container *ngSwitchCase="'Error'">
+          <span class="pficon pficon-error"></span> Published version {{ integration.deploymentVersion }}
+        </ng-container>
+      </ng-container>
       <dl class="dl-horizontal">
         <dt>
           Description:
@@ -75,64 +89,96 @@
           </p>
         </dd>
       </dl>
-      <ng-container *ngIf="integrationDeployments$ | async; let integrationDeployments">
-        <syndesis-loading [loading]="integrationDeployments === undefined">
-          <dl class="dl-horizontal">
-            <dt>
-              History:
-            </dt>
-            <dd>
-              <pfng-list [items]="integrationDeployments"
-                         [config]="deploymentListConfig"
-                         [itemTemplate]="itemTemplate"
-                         [expandTemplate]="expandTemplate"
-                         [actionTemplate]="actionTemplate">
-                <ng-template #itemTemplate
-                             let-deployment="item"
-                             let-index="index">
-                  <div class="list-pf-left">
-                    <ng-container *ngIf="deployment.version === (integration.deploymentId || integration.version)">
-                      <span class="pficon pficon-ok list-pf-icon-small"></span>
-                    </ng-container>
-                  </div>
-                  <div class="list-pf-content-wrapper">
-                    <div class="list-pf-main-content">
-                      <div class="list-pf-title">
-                        Version: {{ deployment.version }}
+      <ng-container *ngIf="integration.draft">
+        <dl class="dl-horizontal">
+          <dt>
+            Draft:
+          </dt>
+          <dd>
+            <div class="list-pf">
+              <div class="list-pf-item">
+                <div class="list-pf-container">
+                  <div class="list-pf-content list-pf-content-flex">
+                    <div class="list-pf-left">
+                    </div>
+                    <div class="list-pf-content-wrapper">
+                      <div class="list-pf-main-content">
+                        <div class="list-pf-title">
+                          <i>Draft</i>
+                        </div>
                       </div>
                     </div>
-                  </div>
-                  <div class="list-pf-additional-content">
-                    <span>Last published on {{ deployment.lastUpdated | date }}</span>
-                    <span>{{ deployment.timesUsed | i18nPlural:usesMapping }} </span>
-                  </div>
-                </ng-template>
-                <ng-template #expandTemplate let-deployment="item" let-index="index">
-                  <div class="row">
-                    <div class="col-md-6">
-                      <dl class="dl-horizontal">
-                        <dt>Created at:</dt>
-                        <dd>{{ deployment.createdDate | date }}</dd>
-                      </dl>
+                    <div class="list-pf-additional-content">
+
                     </div>
-                    <div class="col-md-6">
-                      <dl class="dl-horizontal">
-                        <dt></dt>
-                        <dd></dd>
-                      </dl>
+                    <div class="list-pf-actions">
+                      <pfng-action [config]="draftConfig"
+                                   (onActionSelect)="draftAction($event)">
+                      </pfng-action>
                     </div>
                   </div>
-                </ng-template>
-                <ng-template #actionTemplate let-deployment="item" let-index="index">
-                  <pfng-action class="list-pf-actions"
-                               [config]="deploymentActionConfigs[deployment.id]"
-                               (onActionSelect)="deploymentAction($event, deployment)">
-                  </pfng-action>
-                </ng-template>
-              </pfng-list>
-            </dd>
-          </dl>
-        </syndesis-loading>
+                </div>
+              </div>
+            </div>
+          </dd>
+        </dl>
+      </ng-container>
+      <ng-container *ngIf="integration.deployments">
+        <dl class="dl-horizontal">
+          <dt>
+            History:
+          </dt>
+          <dd>
+            <pfng-list [items]="integration.deployments"
+                        [config]="deploymentListConfig"
+                        [itemTemplate]="itemTemplate"
+                        [expandTemplate]="expandTemplate"
+                        [actionTemplate]="actionTemplate">
+              <ng-template #itemTemplate
+                            let-deployment="item"
+                            let-index="index">
+                <div class="list-pf-left">
+                  <ng-container *ngIf="deployment.version === integration.deploymentVersion">
+                    <span class="pficon pficon-ok list-pf-icon-small"></span>
+                  </ng-container>
+                </div>
+                <div class="list-pf-content-wrapper">
+                  <div class="list-pf-main-content">
+                    <div class="list-pf-title">
+                      Version: {{ deployment.version }}
+                    </div>
+                  </div>
+                </div>
+                <div class="list-pf-additional-content">
+                  <span *ngIf="deployment.updatedAt">Last published on {{ deployment.updatedAt | date }}</span>
+                  <span>{{ deployment.timesUsed | i18nPlural:usesMapping }} </span>
+                </div>
+              </ng-template>
+              <ng-template #expandTemplate let-deployment="item" let-index="index">
+                <div class="row">
+                  <div class="col-md-6">
+                    <dl class="dl-horizontal" *ngIf="deployment.createdAt">
+                      <dt>Created at:</dt>
+                      <dd>{{ deployment.createdAt | date }}</dd>
+                    </dl>
+                  </div>
+                  <div class="col-md-6">
+                    <dl class="dl-horizontal">
+                      <dt></dt>
+                      <dd></dd>
+                    </dl>
+                  </div>
+                </div>
+              </ng-template>
+              <ng-template #actionTemplate let-deployment="item" let-index="index">
+                <pfng-action class="list-pf-actions"
+                              [config]="deploymentActionConfigs[deployment.id]"
+                              (onActionSelect)="deploymentAction($event, deployment)">
+                </pfng-action>
+              </ng-template>
+            </pfng-list>
+          </dd>
+        </dl>
       </ng-container>
       <dl class="dl-horizontal" *ngIf="loggingEnabled">
         <dt>

--- a/app/ui/src/app/integration/detail-page/detail.component.ts
+++ b/app/ui/src/app/integration/detail-page/detail.component.ts
@@ -10,7 +10,6 @@ import { Integration,
   Step,
   PUBLISHED,
   UNPUBLISHED,
-  DRAFT,
   IntegrationOverview,
   IntegrationActionsService,
   IntegrationSupportService,
@@ -65,6 +64,7 @@ export class IntegrationDetailComponent implements OnInit, OnDestroy {
   };
   deploymentListConfig: ListConfig;
   deploymentActionConfigs: { [id: string]: ActionConfig } = {};
+  draftConfig: ActionConfig;
   currentDeployment: IntegrationDeployment;
 
   constructor(
@@ -109,9 +109,10 @@ export class IntegrationDetailComponent implements OnInit, OnDestroy {
   }
 
   attributeUpdated(attr: string, value: string) {
-    this.integration[attr] = value;
+    const attributes = {};
+    attributes[attr] = value;
     this.store
-      .update(<any> this.integration)
+      .patch( <any> this.integration, attributes)
       .toPromise()
       .then((update: Integration) => {
         this.notificationService.popNotification({
@@ -129,35 +130,40 @@ export class IntegrationDetailComponent implements OnInit, OnDestroy {
       });
   }
 
-  deploymentAction(event, deployment) {
+  draftAction(event) {
+    switch (event.id) {
+      case 'publish':
+        this.integrationActionsService.requestAction('publish', <any> this.integration);
+        break;
+      case 'edit':
+        this.integrationActionsService.requestAction('edit', <any> this.integration);
+        break;
+      default:
+        break;
+    }
+  }
+
+  deploymentAction(event, deployment: IntegrationDeployment) {
     switch (event.id) {
       case REPLACE_DRAFT:
-        {
-          /*
-          const integration = { ...this.integration };
-          integration.steps = deployment.spec.steps;
-          //integration.desiredStatus = DRAFT;
-          this.integrationActionsService.requestAction('replaceDraft', integration);
-          */
-        }
+        this.integrationActionsService.requestAction('replaceDraft', this.integration, deployment);
         break;
       case CREATE_DRAFT:
         // TODO doesn't this just mean edit?
         break;
       case STOP_INTEGRATION:
-        /*
-        this.integrationActionsService.requestAction('deactivate', this.integration);
-        */
+        this.integrationActionsService.requestAction('deactivate', <any> this.integration);
         break;
       case PUBLISH:
-        /*
         {
-          const integration = { ...this.integration };
-          integration.steps = deployment.spec.steps;
-          //this.integration.desiredStatus = PUBLISHED;
-          this.integrationActionsService.requestAction('publish', integration);
+          const integration = {
+            integrationVersion: this.integration.version,
+            id: this.integration.id,
+            name: this.integration.name,
+            version: deployment.version
+           };
+          this.integrationActionsService.requestAction('publish', <any> integration);
         }
-        */
         break;
       default:
     }
@@ -168,6 +174,17 @@ export class IntegrationDetailComponent implements OnInit, OnDestroy {
   }
 
   ngOnInit() {
+    this.draftConfig = {
+      primaryActions: [
+        {... publish, ...{ styleClass: 'btn btn-default primary-action' } },
+        {
+          id: 'edit',
+          title: 'Edit',
+          styleClass: 'btn btn-default primary-action',
+          tooltip: 'Edit this draft'
+        } as PFAction,
+      ]
+    } as ActionConfig;
     this.deploymentListConfig = {
       selectItems: false,
       showCheckbox: false,
@@ -184,6 +201,26 @@ export class IntegrationDetailComponent implements OnInit, OnDestroy {
         this.integrationSubscription = this.integration$.subscribe((integration: IntegrationOverview) => {
           this.loading = false;
           this.integration = integration;
+          this.deploymentActionConfigs = {};
+          for (const deployment of this.integration.deployments) {
+            const actionConfig = {
+              primaryActions: [],
+              moreActions: [],
+              moreActionsVisible: true,
+              moreActionsDisabled: false
+            } as ActionConfig;
+            actionConfig.moreActions.push(replaceDraft);
+            if (deployment.version === integration.deploymentVersion) {
+              if (integration.currentState === PUBLISHED) {
+                actionConfig.moreActions.push(stopIntegration);
+              } else {
+                actionConfig.moreActions.push(publish);
+              }
+            } else {
+              actionConfig.moreActions.push(publish);
+            }
+            this.deploymentActionConfigs[deployment.id] = actionConfig;
+          }
         });
       });
   }

--- a/app/ui/src/app/integration/edit-page/action-configure/action-configure.component.ts
+++ b/app/ui/src/app/integration/edit-page/action-configure/action-configure.component.ts
@@ -173,17 +173,8 @@ export class IntegrationConfigureActionComponent implements OnInit, OnDestroy {
       .then(descriptor => {
         this.initForm(position, page, descriptor);
       })
-      .catch(response => {
-        log.debug('Error response: ' + JSON.stringify(response, undefined, 2));
-        try {
-          const error = JSON.parse(response[ '_body' ]);
-          this.initForm(position, page, undefined, error);
-        } catch (err) {
-          /* Bailout at this point... */
-          this.initForm(position, page, undefined, {
-            message: err
-          });
-        }
+      .catch(error => {
+        this.initForm(position, page, undefined, error);
       });
   }
 

--- a/app/ui/src/app/integration/integration-actions-provider.service.ts
+++ b/app/ui/src/app/integration/integration-actions-provider.service.ts
@@ -1,10 +1,12 @@
 import { ApplicationRef, Injectable } from '@angular/core';
 import { ActivatedRoute, Router } from '@angular/router';
 import { Integration,
+  IntegrationDeployment,
+  IntegrationOverview,
+  DeploymentOverview,
   IntegrationActionsService,
   IntegrationSupportService,
   Step,
-  DRAFT,
   PENDING,
   PUBLISHED,
   UNPUBLISHED } from '@syndesis/ui/platform';
@@ -17,8 +19,8 @@ import { saveAs } from 'file-saver';
 
 @Injectable()
 export class IntegrationActionsProviderService extends IntegrationActionsService {
-  currentAction: string = undefined;
-  selectedIntegration: Integration = undefined;
+  currentAction: string;
+  selectedIntegration: Integration | IntegrationOverview ;
   modalTitle: string;
   modalMessage: string;
 
@@ -34,19 +36,19 @@ export class IntegrationActionsProviderService extends IntegrationActionsService
     super();
   }
 
-  canActivate(integration: Integration) {
+  canActivate(integration: Integration | IntegrationOverview) {
     // TODO: false if integration.version == lastDeloyed.version && is active.
     return true; // integration.currentStatus === UNPUBLISHED || integration.currentStatus === DRAFT;
   }
 
-  canDeactivate(integration: Integration) {
+  canDeactivate(integration: Integration | IntegrationOverview) {
     // TODO: true if lastDeloyed is active.
     return true; //integration.currentStatus === PUBLISHED || integration.currentStatus === PENDING;
   }
 
   //----- Actions ------------------->>
 
-  requestAction(action: string, integration: Integration) {
+  requestAction(action: string, integration: Integration | IntegrationOverview, deployment?: IntegrationDeployment | DeploymentOverview) {
     let request, header, message, danger, reason;
     switch (action) {
       case 'createIntegration':
@@ -66,7 +68,7 @@ export class IntegrationActionsProviderService extends IntegrationActionsService
         message = 'Replacing the current draft of the integration';
         danger = 'Failed to update integration draft';
         reason = 'Error updating integration';
-        request = this.requestReplaceDraft(integration);
+        request = this.requestReplaceDraft(integration, deployment);
         break;
       case 'publish':
         header = 'Publishing deployment';
@@ -105,7 +107,7 @@ export class IntegrationActionsProviderService extends IntegrationActionsService
     return request.then(
       modal =>
         modal.result
-          ? this.doAction(action, integration)
+          ? this.doAction(action, integration, deployment)
               .then(_ =>
                 this.notificationService.popNotification({
                   type: NotificationType.SUCCESS,
@@ -133,10 +135,10 @@ export class IntegrationActionsProviderService extends IntegrationActionsService
     return this.modalMessage || '';
   }
 
-  doAction(action: string, integration: Integration) {
+  doAction(action: string, integration: Integration | IntegrationOverview, deployment?: IntegrationDeployment | DeploymentOverview) {
     switch (action) {
       case 'replaceDraft':
-        return this.replaceDraftAction(integration);
+        return this.replaceDraftAction(integration, deployment);
       case 'activate':
       case 'publish':
         return this.activateAction(integration);
@@ -150,19 +152,19 @@ export class IntegrationActionsProviderService extends IntegrationActionsService
   }
 
   //-----  Activate/Deactivate ------------------->>
-  requestReplaceDraft(integration: Integration) {
+  requestReplaceDraft(integration: Integration | IntegrationOverview, deployment: IntegrationDeployment | DeploymentOverview) {
     this.selectedIntegration = integration;
     return this.showModal('replaceDraft');
   }
 
-  requestPublish(integration: Integration) {
+  requestPublish(integration: Integration | IntegrationOverview) {
     this.selectedIntegration = integration;
     return this.showModal('publish');
   }
 
   // TODO: Refactor into single method for both cases
   // Open modal to confirm activation
-  requestActivate(integration: Integration) {
+  requestActivate(integration: Integration | IntegrationOverview) {
     log.debugc(
       () =>
         'Selected integration for activation: ' +
@@ -173,7 +175,7 @@ export class IntegrationActionsProviderService extends IntegrationActionsService
   }
 
   // Open modal to confirm deactivation
-  requestDeactivate(integration: Integration) {
+  requestDeactivate(integration: Integration | IntegrationOverview) {
     log.debugc(
       () =>
         'Selected integration for deactivation: ' +
@@ -184,7 +186,7 @@ export class IntegrationActionsProviderService extends IntegrationActionsService
   }
 
   // Open modal to confirm delete
-  requestDelete(integration: Integration) {
+  requestDelete(integration: Integration | IntegrationOverview) {
     log.debugc(
       () =>
         'Selected integration for delete: ' + JSON.stringify(integration['id'])
@@ -195,39 +197,45 @@ export class IntegrationActionsProviderService extends IntegrationActionsService
 
   // TODO: Refactor into single method for both cases
   // Actual activate/deactivate action once the user confirms
-  activateAction(integration: Integration): Promise<any> {
+  activateAction(integration: Integration | IntegrationOverview): Promise<any> {
     log.debugc(
       () =>
         'Selected integration for activation: ' +
         JSON.stringify(integration['id'])
     );
-    return this.integrationSupportService.deploy(integration).toPromise();
+    return this.integrationSupportService.deploy(<any> integration).toPromise();
   }
 
   // Actual activate/deactivate action once the user confirms
-  deactivateAction(integration: Integration): Promise<any> {
+  deactivateAction(integration: Integration | IntegrationOverview): Promise<any> {
     log.debugc(
       () =>
         'Selected integration for deactivation: ' +
         JSON.stringify(integration['id'])
     );
-    return this.integrationSupportService.undeploy(integration).toPromise();
+    return this.integrationSupportService.undeploy(<any> integration).toPromise();
   }
 
   // Actual delete action once the user confirms
-  deleteAction(integration: Integration): Promise<any> {
+  deleteAction(integration: Integration | IntegrationOverview): Promise<any> {
     log.debugc(
       () =>
         'Selected integration for delete: ' + JSON.stringify(integration['id'])
     );
     return this.store
-      .delete(integration)
+      .delete(<any> integration)
       .take(1)
       .toPromise();
   }
 
-  replaceDraftAction(integration: Integration): Promise<any> {
-    return this.store.update(integration, true).take(1).toPromise();
+  replaceDraftAction(integration: Integration | IntegrationOverview, deployment: IntegrationDeployment | DeploymentOverview): Promise<any> {
+      return this.integrationSupportService.getDeployment(integration.id, deployment.version.toString())
+        .map(_deployment => {
+          this.store.patch(<any>integration, {
+            steps: _deployment.spec.steps,
+            draft: true
+          });
+        }).toPromise();
   }
 
   //-----  Icons ------------------->>

--- a/app/ui/src/app/integration/integration-support-provider.service.ts
+++ b/app/ui/src/app/integration/integration-support-provider.service.ts
@@ -1,7 +1,6 @@
 import { Injectable } from '@angular/core';
 import { Observable } from 'rxjs/Observable';
-import {
-  Action,
+import { Action,
   Connection,
   Activity,
   Integration,
@@ -12,7 +11,8 @@ import {
   IntegrationStatus,
   IntegrationSupportService,
   ApiHttpService,
-  UNPUBLISHED
+  UNPUBLISHED,
+  PUBLISHED
 } from '@syndesis/ui/platform';
 import { EventsService } from '@syndesis/ui/store';
 import { integrationSupportEndpoints } from './integration-support.api';
@@ -37,7 +37,16 @@ export class IntegrationSupportProviderService extends IntegrationSupportService
     return Observable.merge(
       this.getOverview(id),
       this.eventsService.changeEvents
-        .filter(event => event.id === id)
+        .filter(event => {
+          switch (event.kind) {
+            case 'integration':
+              return event.id === id;
+            case 'integration-deployment':
+              return event.id.startsWith(id);
+            default:
+              return false;
+          }
+        })
         .flatMap(event => this.getOverview(id))
     );
   }
@@ -50,7 +59,7 @@ export class IntegrationSupportProviderService extends IntegrationSupportService
     return Observable.merge(
       this.getOverviews(),
       this.eventsService.changeEvents
-        .filter(event => event.kind === 'integration')
+        .filter(event => event.kind === 'integration' || event.kind === 'integration-deployment')
         .flatMap(event => this.getOverviews())
     );
   }
@@ -59,12 +68,20 @@ export class IntegrationSupportProviderService extends IntegrationSupportService
     return this.apiHttpService.setEndpointUrl(integrationSupportEndpoints.deployments, { id }).get();
   }
 
-  publishIntegration(integration: Integration): Observable<any> {
-    return this.apiHttpService.setEndpointUrl(integrationSupportEndpoints.publish, { id: integration.id }).post(integration);
-  }
-
-  deploy(integration: Integration): Observable<any> {
-    return this.apiHttpService.setEndpointUrl(integrationSupportEndpoints.publish, { id: integration.id }).put({});
+  deploy(integration: Integration | IntegrationDeployment): Observable<any> {
+    let url, state, method;
+    if ('integrationVersion' in integration) {
+      // it's an IntegrationDeployment
+      url = integrationSupportEndpoints.updateState;
+      state = { targetState: PUBLISHED };
+      method = 'post';
+    } else {
+      // it's an Integration
+      url = integrationSupportEndpoints.publish;
+      state = { };
+      method = 'put';
+    }
+    return this.apiHttpService.setEndpointUrl(url, { id: integration.id, version: integration.version })[method](state);
   }
 
   undeploy(integration: Integration): Observable<any> {

--- a/app/ui/src/app/platform/types/integration/integration-actions.service.ts
+++ b/app/ui/src/app/platform/types/integration/integration-actions.service.ts
@@ -1,12 +1,14 @@
 import { Injectable } from '@angular/core';
 
-import { Integration, Step } from '@syndesis/ui/platform';
+import { Integration, IntegrationOverview, IntegrationDeployment, DeploymentOverview, Step } from '@syndesis/ui/platform';
 
 @Injectable()
 export abstract class IntegrationActionsService {
   abstract canActivate(integration: Integration): boolean;
   abstract canDeactivate(integration: Integration): boolean;
-  abstract requestAction(action: string, integration: Integration);
+  abstract requestAction(action: string,
+                         integration: Integration | IntegrationOverview,
+                         deployment?: IntegrationDeployment | DeploymentOverview);
   abstract getStart(integration: Integration): Step;
   abstract getFinish(integration: Integration): Step;
   abstract getModalTitle(): string;

--- a/app/ui/src/app/platform/types/integration/integration-support.service.ts
+++ b/app/ui/src/app/platform/types/integration/integration-support.service.ts
@@ -44,12 +44,6 @@ export abstract class IntegrationSupportService {
    */
   abstract watchOverviews(): Observable<IntegrationOverviews>;
 
-  /**
-   * Publish an existing integration
-   * @param integration
-   */
-  abstract publishIntegration(integration: Integration): Observable<any>;
-
   abstract deploy(integration: Integration): Observable<any>;
 
   abstract undeploy(integration: Integration): Observable<any>;

--- a/app/ui/src/app/platform/types/integration/integration.models.ts
+++ b/app/ui/src/app/platform/types/integration/integration.models.ts
@@ -16,7 +16,6 @@ export class Step implements BaseEntity {
 
 export type Steps = Array<Step>;
 
-export const DRAFT = 'Draft';
 export const PENDING = 'Pending';
 export const PUBLISHED = 'Published';
 export const UNPUBLISHED = 'Unpublished';
@@ -70,7 +69,6 @@ export interface IntegrationDeployment extends BaseEntity {
 export type IntegrationDeployments = Array<IntegrationDeployment>;
 
 export interface IntegrationOverview extends BaseEntity {
-  version: number;
   name: string;
   tags: Array<string>;
   description?: string;
@@ -79,6 +77,8 @@ export interface IntegrationOverview extends BaseEntity {
   currentState: IntegrationStatus;
   targetState: IntegrationStatus;
   statusMessage?: string;
+  deploymentVersion: number;
+  version: number;
 }
 export type IntegrationOverviews = Array<IntegrationOverview>;
 

--- a/app/ui/src/app/store/entity/entity.store.ts
+++ b/app/ui/src/app/store/entity/entity.store.ts
@@ -261,6 +261,10 @@ export abstract class AbstractStore<
     return deleted.share();
   }
 
+  patch(entity: T, attributes: any): Observable<any> {
+    return this.service.patch(entity.id, attributes);
+  }
+
   private massageError(error: any) {
     let errorMessage: any;
     switch (typeof error) {

--- a/app/ui/src/app/store/entity/rest.service.ts
+++ b/app/ui/src/app/store/entity/rest.service.ts
@@ -1,7 +1,7 @@
 import { Restangular } from 'ngx-restangular';
 import { Observable } from 'rxjs/Observable';
 
-import { BaseEntity, ApiHttpService } from '@syndesis/ui/platform';
+import { BaseEntity } from '@syndesis/ui/platform';
 
 export abstract class RESTService<T extends BaseEntity, L extends Array<T>> {
   protected constructor(
@@ -27,5 +27,9 @@ export abstract class RESTService<T extends BaseEntity, L extends Array<T>> {
 
   delete(obj: T) {
     return this.restangularService.one(obj.id).remove();
+  }
+
+  patch(id: string, attributes: any) {
+    return this.restangularService.one(id).patch(attributes);
   }
 }

